src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -166,16 +170,85 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
     };
 
     handleFetchError = (error: Object): void => {
-        const { onUpgradeClick } = this.props;
+        const { currentUser, onUpgradeClick, showUpsell, showUpsellWithPicture } = this.props;
         const shouldShowVersionErrorWithUpsell = !!onUpgradeClick && error.status === 403;
-        this.setState({
-            error: shouldShowVersionErrorWithUpsell ? messages.versionNotAvailable : messages.versionFetchError,
-            errorTitle: shouldShowVersionErrorWithUpsell ? messages.versionAccessError : messages.versionServerError,
-            isLoading: false,
-            isWatermarked: false,
-            versionCount: 0,
-            versions: [],
-        });
+        const shouldShowVersions = showUpsell || showUpsellWithPicture;
+
+        if (!shouldShowVersions) {
+            this.setState({
+                error: shouldShowVersionErrorWithUpsell ? messages.versionNotAvailable : messages.versionFetchError,
+                errorTitle: shouldShowVersionErrorWithUpsell
```

Comment:
Do we now have three separate upsell experiences? Can we reduce this to one?

Reply:
@jstoffan There are three different ones for the experiment. I am going to try to eliminate how many like flags we're sending over, but the point of the experiment is to try out three separate upsell experiences, yes